### ISSUE TITLE
dask-ml make_classification keywords do not work

### ISSUE THREAD

**Issue by js3711** [dask/dask-ml#323]
Supplying the following keywords to dask_ml.datasets.make_classification does not appear to have the expected result.

n_samples, n_informative, n_redundant, n_features

Take a look at the images of the correlation matrix between the features when comparing scikit-learn's make_classification to dask_ml's make_classification.
```python
import dask_ml.datasets as dask_datasets
import sklearn.datasets as sk_datasets
```

```python
def make_dataset_and_compute_correlation(func, **kwargs):
    X, y = func(**kwargs) 
    ddf_features = dd.from_array(X)
    
    corr = ddf_features.corr().compute()
    
    return corr
```

```python
dask_corr = make_dataset_and_compute_correlation(dask_datasets.make_classification, 
                                                 n_samples=10000, n_informative=12, 
                                                 n_redundant=18, n_features=30, 
                                                 chunks=100)
```
![dask_ml_correlation](https://user-images.githubusercontent.com/15849929/43396566-f46c90b6-93cf-11e8-9101-c74d9cd16734.png)

vs

```python
sk_corr = make_dataset_and_compute_correlation(sk_datasets.make_classification, 
                                               n_samples=10000, n_informative=12, 
                                               n_redundant=18, n_features=30)
```
![sklearn_correlation](https://user-images.githubusercontent.com/15849929/43396586-0096ab4c-93d0-11e8-8b9a-c7c9aaa6ce7d.png)

### Dependencies
<details>
# Name                    Version                   Build  Channel<br>
appdirs                   1.4.3            py36h28b3542_0  <br>
appnope                   0.1.0            py36hf537a9a_0  <br>
asn1crypto                0.24.0                   py36_0  <br>
attrs                     18.1.0                   py36_0  <br>
automat                   0.7.0                    py36_0  <br>
backcall                  0.1.0                    py36_0  <br>
blas                      1.0                         mkl  <br>
bleach                    2.1.3                    py36_0  <br>
bokeh                     0.13.0                   py36_0  <br>
ca-certificates           2018.03.07                    0  <br>
cairo                     1.14.12              hc4e6be7_4  <br>
certifi                   2018.4.16                py36_0  <br>
cffi                      1.11.5           py36h342bebf_0  <br>
chardet                   3.0.4                     <pip><br>
click                     6.7              py36hec950be_0  <br>
cloudpickle               0.5.3                    py36_0  <br>
constantly                15.1.0           py36h28b3542_0  <br>
cryptography              2.2.2            py36h1de35cc_0  <br>
cycler                    0.10.0           py36hfc81398_0  <br>
cytoolz                   0.9.0.1          py36h1de35cc_1  <br>
dask                      0.18.2                   py36_0  <br>
dask-core                 0.18.2                   py36_0  <br>
dask-glm                  0.1.0                    py36_0  <br>
dask-ml                   0.7.0            py36h1de35cc_0  <br>
dask-searchcv             0.2.0                    py36_0  <br>
decorator                 4.3.0                    py36_0  <br>
distributed               1.22.0                   py36_0  <br>
entrypoints               0.2.3                    py36_2  <br>
expat                     2.2.5                hb8e80ba_0  <br>
fontconfig                2.13.0               h5d5b041_1  <br>
freetype                  2.9.1                hb4e5f40_0  <br>
fribidi                   1.0.4                h1de35cc_0  <br>
gettext                   0.19.8.1             h15daf44_3  <br>
glib                      2.56.1               h35bc53a_0  <br>
graphite2                 1.3.11               h2098e52_2  <br>
graphviz                  2.40.1               hefbbd9a_2  <br>
harfbuzz                  1.7.6                hb8d4a28_3  <br>
heapdict                  1.0.0                    py36_2  <br>
html5lib                  1.0.1                    py36_0  <br>
hyperlink                 18.0.0                   py36_0  <br>
icu                       58.2                 h4b95b61_1  <br>
idna                      2.7                      py36_0  <br>
incremental               17.5.0                   py36_0  <br>
intel-openmp              2018.0.3                      0  <br>
ipykernel                 4.8.2                    py36_0  <br>
ipython                   6.4.0                    py36_1  <br>
ipython_genutils          0.2.0            py36h241746c_0  <br>
ipywidgets                7.3.0                    py36_0  <br>
jedi                      0.12.1                   py36_0  <br>
jinja2                    2.10                     py36_0  <br>
jpeg                      9b                   he5867d9_2  <br>
jsonschema                2.6.0            py36hb385e00_0  <br>
jupyter_client            5.2.3                    py36_0  <br>
jupyter_core              4.4.0                    py36_0  <br>
jupyterlab                0.32.1                   py36_0  <br>
jupyterlab_launcher       0.10.5                   py36_0  <br>
kiwisolver                1.0.1            py36h0a44026_0  <br>
libcxx                    4.0.1                h579ed51_0  <br>
libcxxabi                 4.0.1                hebd6815_0  <br>
libedit                   3.1.20170329         hb402a30_2  <br>
libffi                    3.2.1                h475c297_4  <br>
libgfortran               3.0.1                h93005f0_2  <br>
libiconv                  1.15                 hdd342a3_7  <br>
libpng                    1.6.34               he12f830_0  <br>
libsodium                 1.0.16               h3efe00b_0  <br>
libtiff                   4.0.9                hcb84e12_1  <br>
libxml2                   2.9.8                hab757c2_1  <br>
locket                    0.2.0            py36hca03003_1  <br>
markupsafe                1.0              py36h1de35cc_1  <br>
matplotlib                2.2.2            py36hbf02d85_2  <br>
mistune                   0.8.3            py36h1de35cc_1  <br>
mkl                       2018.0.3                      1  <br>
mkl_fft                   1.0.2            py36h6b9c3cc_0  <br>
mkl_random                1.0.1            py36h5d10147_1  <br>
msgpack-python            0.5.6            py36h04f5b5a_0  <br>
multipledispatch          0.5.0                    py36_0  <br>
nbconvert                 5.3.1                    py36_0  <br>
nbformat                  4.4.0            py36h827af21_0  <br>
ncurses                   6.1                  h0a44026_0  <br>
networkx                  2.1                      py36_0  <br>
notebook                  5.6.0                    py36_0  <br>
numpy                     1.14.5           py36h648b28d_4  <br>
numpy-base                1.14.5           py36ha9ae307_4  <br>
openssl                   1.0.2o               h26aff7b_0  <br>
packaging                 17.1                     py36_0  <br>
pandas                    0.23.3           py36h6440ff4_0  <br>
pandoc                    2.2.1                h1a437c5_0  <br>
pandocfilters             1.4.2                    py36_1  <br>
pango                     1.42.1               he2d0c7e_2  <br>
parso                     0.3.1                    py36_0  <br>
partd                     0.3.8            py36hf5c4cb8_0  <br>
pcre                      8.42                 h378b8a2_0  <br>
pexpect                   4.6.0                    py36_0  <br>
pickleshare               0.7.4            py36hf512f8e_0  <br>
pip                       10.0.1                   py36_0  <br>
pixman                    0.34.0               hca0a616_3  <br>
plotly                    3.0.0rc11                 <pip><br>
prometheus_client         0.2.0                    py36_0  <br>
prompt_toolkit            1.0.15           py36haeda067_0  <br>
psutil                    5.4.6            py36h1de35cc_0  <br>
ptyprocess                0.6.0                    py36_0  <br>
pyasn1                    0.4.3                    py36_0  <br>
pyasn1-modules            0.2.2                    py36_0  <br>
pycparser                 2.18                     py36_1  <br>
pygments                  2.2.0            py36h240cd3f_0  <br>
pygraphviz                1.3              py36h1de35cc_1  <br>
pyopenssl                 18.0.0                   py36_0  <br>
pyparsing                 2.2.0                    py36_1  <br>
python                    3.6.6                hc167b69_0  <br>
python-dateutil           2.7.3                    py36_0  <br>
python.app                2                        py36_8  <br>
pytz                      2018.5                   py36_0  <br>
pyyaml                    3.13             py36h1de35cc_0  <br>
pyzmq                     17.0.0           py36h1de35cc_3  <br>
readline                  7.0                  hc1231fa_4  <br>
requests                  2.19.1                    <pip><br>
retrying                  1.3.3                     <pip><br>
scikit-learn              0.19.1           py36hf9f1f73_0  <br>
scipy                     1.1.0            py36hf1f7d93_0  <br>
send2trash                1.5.0                    py36_0  <br>
service_identity          17.0.0           py36h28b3542_0  <br>
setuptools                39.2.0                   py36_0  <br>
simplegeneric             0.8.1                    py36_2  <br>
six                       1.11.0                   py36_1  <br>
sortedcontainers          2.0.4                    py36_0  <br>
sqlite                    3.24.0               ha441bb4_0  <br>
tblib                     1.3.2            py36hda67792_0  <br>
terminado                 0.8.1                    py36_1  <br>
testpath                  0.3.1            py36h625a49b_0  <br>
tk                        8.6.7                h35a86e2_3  <br>
toolz                     0.9.0                    py36_0  <br>
tornado                   5.0.2            py36h1de35cc_0  <br>
traitlets                 4.3.2            py36h65bd3ce_0  <br>
twisted                   17.5.0                   py36_0  <br>
urllib3                   1.23                      <pip><br>
wcwidth                   0.1.7            py36h8c6ec74_0  <br>
webencodings              0.5.1                    py36_1  <br>
wheel                     0.31.1                   py36_0  <br>
widgetsnbextension        3.3.0                    py36_0  <br>
xz                        5.2.4                h1de35cc_4  <br>
yaml                      0.1.7                hc338f04_2  <br>
zeromq                    4.2.5                h0a44026_0  <br>
zict                      0.1.3                    py36_0  <br>
zlib                      1.2.11               hf3cbc9b_2  <br>
zope                      1.0                      py36_0  <br>
zope.interface            4.5.0            py36h1de35cc_0 

</details>

**Comment by js3711** [comment]
[![Binder](https://mybinder.org/badge.svg)](https://mybinder.org/v2/gh/js3711/dask_ml_323/master)

**Comment by TomAugspurger** [comment]
Currently n_redundant, n_repeated, n_clusters_per_class, weights, flip_y, class_sep, hypercube, shift, and shuffle have no effect in `dask_ml.datasets.make_classification`.

Would welcome any improvements here (docs or data generate).

**Comment by js3711** [comment]
I might recommend removing the keywords or throwing an error/warning if they are used until the logic has actually been implemented.

**Comment by TomAugspurger** [comment]
I think our (unstated) policy is to match the signature, but raise if the keyword is specified but not implemented. So a PR checking for that would also be most welcome.

**Comment by mrocklin** [comment]
It seems reasonable to me to not match the signature if we genuinely don't
support the keyword

**Comment by mrocklin** [comment]
Scikit-Learn functions often have many keywords.  Raising informative
NotImplementedErrors for all of them sounds potentially painful

**Comment by mmccarty** [comment]
@TomAugspurger @mrocklin  Just a heads up that I'm working on closing the gaps here.
